src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc

```diff
@@ -0,0 +1,253 @@
+= Hawkular and Grafana Out of the Box
+Joel Takvorian
+2017-05-31
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, grafana, docker, vertx
+:figure-caption!:
+
+link:https://github.com/jotak/hawkular-grafana-outofthebox[Hawkular and Grafana Out of the Box] is a new way to quickly setup Hawkular and Grafana together with very little work.
+
+Assuming you already have docker installed and recent enough (you need at least Compose 1.6.0 and Docker Engine 1.10.0),
+you're just 2 lines of YAML edition and a `docker-compose up` away from having Hawkular + Grafana running and pre-configured.
+
+It is NOT intended for production/long-term usage unless you do some extra work to make it secured.
+
+== How does it work?
+
+Let's have a look to the docker-compose file:
+
+[source,yaml]
+.docker-compose.yml
+----
+version: '2' # <1>
+services:
+  hawkular:
+    image: "hawkular/hawkular-services:latest" # <2>
+    ports:
+      - "8080:8080"
+      - "8443:8443"
+      - "9990:9990"
+    # volumes: # <3>
+    #   - /tmp/opt/hawkular/server:/opt/data:Z
+    environment:
+      - CASSANDRA_NODES=hawkular-cassandra
+      - ADMIN_TOKEN
+      - HAWKULAR_AGENT_ENABLE=false # <4>
+  hawkular-cassandra: # <5>
+    image: cassandra:3.0.9
+    environment:
+      - CASSANDRA_START_RPC=true
+  grafana: # <6>
+    image: "hawkular/hawkular-grafana-datasource:latest"
+    ports:
+      - "3000:3000"
+  bootstrap:
+    image: "jotak/hawkular-grafana-bootstrap:latest" # <7>
+    environment:
+      - TENANT=put_your_tenant_here # <8>
+      - DASHBOARDS=put_dashboard_file_names_here # <9>
+----
+<1> Docker-compose format v2 brings some networking improvements that proved to be useful here, that's why you'll need docker 1.10.0+ and docker-compose 1.6.0+.
+<2> Using the official Hawkular Services image. Accessible on http://localhost:8080/hawkular/metrics or http://localhost:8080/hawkular/alerts, with credentials _jdoe_/_password_.
+<3> The volume is commented-out, so data is not persisted if you run `docker-compose down`. To persist data, uncomment these lines and eventually set something different than `/tmp/opt/hawkular/server` as the host directory. Make sure this directory exists on your host.
+<4> The Hawkular agent is disabled by default, which means Hawkular is not self-monitored. You can enable it if you want, and use the `hawkular-self` dashboard.
+<5> The Cassandra node. Note that it's not exposed outside of the compose's subnetwork.
+<6> The Grafana server with pre-installed Hawkular plugin, accessible on http://localhost:3000, with credentials _admin_/_admin_
+<7> This image contains a bootstrap script that configures a Grafana datasource with the given tenant (<8>), which is also injected in dashboard templates defined here (<9>). Those dashboards are sent to Grafana through its REST API. Current available dashboard templates are listed below.
```

Comment:
Those are TENANT and DASHBOARDS in the env section.
@jotak: I think they may only render correctly when you put the `<x>` on the start of a line.